great_expectations/experimental/datasources/data_asset/data_connector/batch_filter.py

```diff
@@ -0,0 +1,260 @@
+from __future__ import annotations
+
+import itertools
+import logging
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional, Union
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations.core.id_dict import IDDict
+from great_expectations.util import is_int
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+
+
+def build_batch_filter(
+    data_connector_query_dict: Optional[
```

Comment:
This function body is copied from `great_expectations/datasource/data_connector/batch_filter.py`. Could you reuse the code there, refactoring if necessary?

Reply:
@billdirks Thank you for catching it -- this whole module is now under experimental, but is not ready for use yet.  Its use is commented out from everywhere in this pull request "with a note to self" -- I will comment it out entirely (but keep the file reference -- commented out -- for now -- so that it can be evolved properly (ZEP-style) in a next effort.  Thanks!

Reply:
@billdirks It is being reused -- in `file_path_data_connect.py` -- the actual BatchFilter module is now under `experimental`.  No refactoring seemed to be necessary just yet.  Thanks.

Reply:
Whe I do a diff ignoring blank lines on this function and the v3 version I see no differences:
```
% diff -B experiment_build_batch_filter.py v3_build_batch_filter.py 
(gx-py3.8) 
%
```
Why can't we just import this?

Reply:
@billdirks Even though we could probably technically import it right now, as the ZEP requirements diverge from V3, it would be costly for us to maintain one object that serves two fundamentally different architectures.  While I do not see the benefit for it, I am happy to discuss.  Thanks.

Reply:
@billdirks I got this done -- and many more!  Thanks for pairing!